ComplementaryScripts/modelCuration/modMetsandSmatrix.m

```diff
@@ -0,0 +1,207 @@
+    currentmetFormula = model.metFormulas(idx_met);
+    currentmetCharges = model.metCharges(idx_met);
+    
+    for j = 1:length(idx_met)
+        %Preliminary check of metBalance/metCharges before change
+        idx_rxn = find(model.S(idx_met(j),:));
+        rxn = model.rxns(idx_rxn);
+        MassChargeresults = checkMassChargeBalance(model,rxn);
+            
+        if isequal(currentmetFormula(j),newFormula(i))
+            %no action required
+        elseif ~isequal(currentmetFormula(j),currentFormula(i)) && ismember(currentFormula(i),'[]') || isequal(currentmetFormula(j),currentFormula(i)) && ~ismember(newFormula(i),'[]')
+            model.metFormulas(idx_met(j)) = newFormula(i);
+            if ~contains(model.metNotes(idx_met(j)),'| metFormula curated (PR #222)')
+                model.metNotes(idx_met(j)) = join([model.metNotes(idx_met(j)),'| metFormula curated (PR #222)']);
+            end
+        elseif ~isequal(currentmetFormula(j),currentFormula(i))
+            warning('error with metFormula matching in modMetsandSmatrix.tsv for %s, idx_met: %d', string(metNames(i)), idx_met(j));
+        end
+        
+        if isequal(currentmetCharges(j),newCharge(i))
+            %no action required
+        elseif (isequal(currentmetCharges(j),currentCharge(i)) && ~isnan(newCharge(i))) || (ismissing(currentmetCharges(j)) && isnan(currentCharge(i)) && ~isnan(newCharge(i)))
+            model.metCharges(idx_met(j)) = newCharge(i);
+            if ~contains(model.metNotes(idx_met(j)),'| metCharge curated (PR #222)')
+                model.metNotes(idx_met(j)) = join([model.metNotes(idx_met(j)),'| metCharge curated (PR #222)']);
+            end
+        elseif ~isequal(currentmetCharges(j),currentCharge(i)) && ~ismissing(currentmetCharges(j))
+            warning('error with metCharges matching in modMetsandSmatrix.tsv for %s, idx_met: %d', string(metNames(i)), idx_met(j));
+        end
+        
+        %Find rxns which are unbalanced before change
+        temp_rxn = MassChargeresults(:,1);
+        temp_notes = MassChargeresults(:,2);
+        temp_idx = find(~ismember(temp_notes,'pass'));
+        unbalancedrxnBefore = temp_rxn(temp_idx);
+        unbalancedNotes = temp_notes(temp_idx);
+        unbalancedBefore = length(find(~ismember(temp_notes,'pass')));
+        
+        arrayidx = find(cellfun('isempty', metResults),1);
+        metResults(arrayidx,1) = cellstr(metNames(i));
+        metResults(arrayidx,2) = {[unbalancedrxnBefore,unbalancedNotes]};
+        metResults(arrayidx,4) = num2cell(unbalancedBefore);
+    end
+end
+
+%Modification of H+/H2O/other metabolite coefficient to balance equation
+temp = erase(updateSmatrix(:,1),'"');
+idx = str2num(char(temp));
+met2 = updateSmatrix(:,2);
+rxn2 = updateSmatrix(:,3);
+[~,idx_rxn2] = ismember(rxn2,model.rxns);
+currentCoef = str2double(updateSmatrix(:,4));
+newCoef = str2double(updateSmatrix(:,5));
+
+for i = 1:length(idx)
+    currentmetCoef = model.S(idx(i,1),idx(i,2));
```

Comment:
try to search the index for reaction/met, not use a certain number. The number may change.

Reply:
Issue has been addressed in commit [53c81d9](https://github.com/SysBioChalmers/yeast-GEM/pull/222/commits/53c81d9e0f79b71a8cdac8834826fa0ead86ea5c)